### PR TITLE
ADR0005: Decide on python code style guide

### LINKED ISSUE #1128: Decide on style guideline
There seems to be agreement to discontinue the Secure Systems Lab style guide and instead point to existing well-established guidelines (see https://github.com/secure-systems-lab/code-style-guidelines/issues/20).

To get us started, I suggest to go with the [Google Python style guide](https://google.github.io/styleguide/pyguide.html), and make decisions about acceptable/desirable deviations as they occur.

### PR DESCRIPTION
**Pull request by lukpueh**
Fixes #1128 

**Description of the changes being introduced by the pull request**:
Document why we chose to henceforth use the Google Python style guide with minimal refinements. 

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: docs/adr/0005-use-google-python-style-guide.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+The Secure Systems Lab code style guide, which has been used for most of the
+code base, has become outdated. Through the upcoming rewrite, we have the
+chance to ignore consistency considerations with existing code style and can
+choose a more standard and up-to-date style guide.
+
+## Decision Drivers
+
+* Flaws in original Secure Systems Lab style guide
+* Curating a complete custom style guide is time consuming
+* Well-established style rules lower contribution barrier
```
Comment (joshuagl) [inline docs/adr/0005-use-google-python-style-guide.md:16]: Nit: You might argue this is already covered by the "lower contribution barrier" driver, but this was a major factor for me:

```suggestion
* Well-established style rules lower contribution barrier
* Custom style is not supported by default in common tooling (i.e. editors
  and linters)
```

File: docs/adr/0005-use-google-python-style-guide.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+* Use custom style guide
+* Use Google style guide with refinements
+
+## Decision Outcome
+
+Chosen option: "Use Google style guide with refinements", because the Google
+style guide is a comprehensive, well-established style guide that is mostly
+based on PEP-8 and was accepted by everyone on the TUF team. There is no need
+to replicate these recommendations. However, we do provide a very slim document
+with additional refinements, in order to emphasize on items the we consider
```
Comment (joshuagl) [inline docs/adr/0005-use-google-python-style-guide.md]: Nit: English language construct.
```suggestion
with additional refinements, in order to emphasize items the we consider
```

**Review by joshuagl — APPROVED** [review]
This looks good, thanks @lukpueh. I pointed out a couple of minor nits, but I'm OK with the ADR as is.

(Apologies for being nit-picky about English.)

**Comment by trishankatdatadog** [comment]
Good idea. Could we also use pylint, flake8, and black?

**Comment by lukpueh** [comment]
Thanks for the comments, you two! No need to apologize,  @joshuagl, I do appreciate language corrections, and our [style guide actually encourages](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#386-punctuation-spelling-and-grammar) them.

**Comment by lukpueh** [comment]
@joshuagl, I squashed your suggestions into my commit. @trishankatdatadog, I added another commit to mention linters and formatters and general style guide usage/transition instructions. Would either of you mind taking another look? Thanks!

**Review by trishankatdatadog — APPROVED** [review]
💯

**Comment by trishankatdatadog** [comment]
Hmm, I do have some reservations regarding hanging vs aligned indents in function arguments, but whatever, I can live with it.

**Comment by lukpueh** [comment]
> Hmm, I do have some reservations regarding hanging vs aligned indents in function arguments, but whatever, I can live with it.

What do you mean? Our [new guide specifically favors aligned indent](https://github.com/secure-systems-lab/code-style-guidelines/blob/master/python.md#indentation-and-line-continuation), which is what you seemed [very passionate above](https://github.com/theupdateframework/tuf/issues/1128#issuecomment-720371661).

**Comment by trishankatdatadog** [comment]
> What do you mean? Our [new guide specifically favors aligned indent](https://github.com/secure-systems-lab/code-style-guidelines/blob/master/python.md#indentation-and-line-continuation), which is what you seemed [very passionate above](https://github.com/theupdateframework/tuf/issues/1128#issuecomment-720371661).

My bad, I missed the tweak to the Google guide. Keep calm and carry on 💯 💯

**Pull request merged**